drivers/sensor/f75303/f75303.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT fintek_f75303
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/sensor/f75303.h>
+#include "f75303.h"
+
+LOG_MODULE_REGISTER(F75303, CONFIG_SENSOR_LOG_LEVEL);
+
+static int f75303_fetch(const struct i2c_dt_spec *i2c,
+			uint8_t off_h, uint8_t off_l, uint16_t *sample)
+{
+	uint8_t val_h;
+	uint8_t val_l;
+	int res;
+
+	res = i2c_reg_read_byte_dt(i2c, off_h, &val_h);
+	if (res) {
+		return res;
+	}
+
+	res = i2c_reg_read_byte_dt(i2c, off_l, &val_l);
+	if (res) {
+		return res;
+	}
+
+	*sample = val_h << 3 | val_l >> 5;
+
+	return 0;
+}
+
+static int f75303_fetch_local(const struct device *dev)
+{
+	struct f75303_data *data = dev->data;
+	const struct f75303_config *config = dev->config;
+
+	return f75303_fetch(&config->i2c,
+			    F75303_LOCAL_TEMP_H,
+			    F75303_LOCAL_TEMP_L,
+			    &data->sample_local);
+}
+
+static int f75303_fetch_remote1(const struct device *dev)
+{
+	struct f75303_data *data = dev->data;
+	const struct f75303_config *config = dev->config;
+
+	return f75303_fetch(&config->i2c,
+			    F75303_REMOTE1_TEMP_H,
+			    F75303_REMOTE1_TEMP_L,
+			    &data->sample_remote1);
+}
+
+static int f75303_fetch_remote2(const struct device *dev)
+{
+	struct f75303_data *data = dev->data;
+	const struct f75303_config *config = dev->config;
+
+	return f75303_fetch(&config->i2c,
+			    F75303_REMOTE2_TEMP_H,
+			    F75303_REMOTE2_TEMP_L,
+			    &data->sample_remote2);
+}
+
+static int f75303_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	int res;
+
+	switch ((uint32_t)chan) {
+	case SENSOR_CHAN_ALL:
+		res = f75303_fetch_local(dev);
+		if (res) {
+			break;
+		}
+		res = f75303_fetch_remote1(dev);
+		if (res) {
+			break;
+		}
+		res = f75303_fetch_remote2(dev);
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
```

Comment:
This sort of thing is exactly why we need to support channel type and index pairs and not just channel type

Reply:
Are there any open issues/proposals to to do this?
This could also be modeled as a [multi-function device](https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/mfd), with each temperature sensor nested under a parent device.  But that seems a little heavyweight.

Reply:
It's being discussed in #61163, though to the best of my knowledge, there is no dedicated issue about this.

Reply:
#13718 #1387 both note the need for multiple channels of the same type